### ISSUE TITLE
Automatic style-checking and/or scalariform rewriting

### ISSUE THREAD

**Issue by non** [non/algebra#58]
I have a branch which adds style-checking and Scalariform.

How do people feel about this? If we are going to do this, it will require one big PR to get all the warnings dealt with and the code reformatted. I seem to recall @johnynek and @avibryant both saying positive things about Scalariform.

Running a test, the only thing I don't love is that:

```scala
tpls.map { case (x, y) =>
  ...
}
```

is rewritten to:

```scala
tpls.map {
  case (x, y) =>
    ...
}
```

But I can live with this if everyone else wants to move in this direction.

**Comment by johnynek** [comment]
yeah, I'm not crazy about that, but automatic uniformity is nice.

Haven't checked lately to see if we can disable that one (and add my pet one: no superfluous `{ }`).
